### PR TITLE
[#173338092] add type validation on activity call

### PR DESCRIPTION
**Pull request by balanza**
This implements encoding on activity call to introduce compiler time check on future input changes

### CODE REVIEW

File: models/eligibility_check.ts [hunk 1]
```diff
@@ -42,11 +41,10 @@ export type NewEligibilityCheck = t.TypeOf<typeof NewEligibilityCheck>;
 function toRetrieved(
   result: DocumentDb.RetrievedDocument
 ): RetrievedEligibilityCheck {
-  return RetrievedEligibilityCheck.decode(result).getOrElseL(err => {
-    throw new Error(
-      `Failed decoding RetrievedEligibilityCheck object: ${readableReport(err)}`
-    );
-  });
+  return {
```
Comment (gunzip) [inline models/eligibility_check.ts:44]: why this change ?

File: DeleteEligibilityCheckActivity/handler.ts [hunk 2]
```diff
@@ -34,9 +39,15 @@ export const getDeleteEligibilityCheckActivityHandler = (
...
-      )
+      DeleteEligibilityCheckActivityInput.decode(input)
+        .mapLeft(
+          err => new Error(`Invalid Activity input: [${readableReport(err)}]`)
+        )
+        .map(
+          fiscalCode =>
+            // pattern strings ain't nonempty strings
+            (fiscalCode as unknown) as NonEmptyString
+        )
```
Comment (BurnedMarshal) [inline DeleteEligibilityCheckActivity/handler.ts:50]:
```suggestion
        .chain(fiscalCode => NonEmptyString.decode(fiscalCode))
        .mapLeft(
          err => new Error(`Invalid Activity input: [${readableReport(err)}]`)
        )
```
